var/spack/repos/builtin/packages/umpire/package.py

```diff
@@ -458,37 +456,56 @@ def initconfig_package_entries(self):
     def cmake_args(self):
         return []
 
-    def test(self):
+    def run_umpire(self, exe, expected):
         """Perform stand-alone checks on the installed package."""
-        if self.spec.satisfies("@:1") or not os.path.isdir(self.prefix.bin):
-            tty.info("Skipping: checks not installed in bin for v{0}".format(self.version))
-            return
-
-        # Run a subset of examples PROVIDED installed
-        # tutorials with readily checkable outputs.
-        checks = {
-            "malloc": ["99 should be 99"],
-            "recipe_dynamic_pool_heuristic": ["in the pool", "releas"],
-            "recipe_no_introspection": ["has allocated", "used"],
-            "strategy_example": ["Available allocators", "HOST"],
-            "tut_copy": ["Copied source data"],
-            "tut_introspection": ["Allocator used is HOST", "size of the allocation"],
-            "tut_memset": ["Set data from HOST"],
-            "tut_move": ["Moved source data", "HOST"],
-            "tut_reallocate": ["Reallocated data"],
-            "vector_allocator": [""],
-        }
-
-        for exe in checks:
-            expected = checks[exe]
-            reason = "test: checking output from {0}".format(exe)
-            self.run_test(
-                exe,
-                [],
-                expected,
-                0,
-                installed=False,
-                purpose=reason,
-                skip_missing=True,
-                work_dir=self.prefix.bin,
-            )
+        if self.spec.satisfies("@:1"):
+            raise SkipTest("Package must be installed as version @1.0.1 or later")
+
+        if os.path.isdir(self.prefix.bin):
+            raise SkipTest(f"{self.prefix.bin} does not eixst")
+
+        exe_run = which(join_path(self.prefix.bin, exe))
+        if exe_run is None:
+            raise SkipTest("Executable not present within directory")
+        out = exe_run(output=str.split, error=str.split)
+        check_outputs(expected, out)
+
+    def test_malloc(self):
+        """Malloc Test"""
+        self.run_umpire("malloc", ["99 should be 99"])
+
+    def test_recipe_dynamic_pool_heuristic(self):
+        """Test heurisitc that uses an allocator for more than just initial allocation"""
```

Comment:
There's a typo here

Reply:
The docstring is a bit long too for a test part description.  See if you can shorten it.